### PR TITLE
[luci] Propagate concat qparam during quantization

### PR DESCRIPTION
**Pull request by jinevening**
This changes PropagateConcatenationQparamPass to a function inside QuantizeWithMinMaxPass

  - To guarantee the validity of graph before/after applying Pass

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to #4004

### CODE REVIEW

**Comment by jinevening** [comment]
This is refactoring of #4064 and #4085.

[This comment](https://github.com/Samsung/ONE/pull/4064#discussion_r483361713) is the reason for the refactoring.

**Review by seanshpark — DISMISSED** [review]
LGTM

File: compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp
Comment (s-barannikov) [inline compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp:609]: Looks redundant in the view of the check on the next line. It could also break a negative test in debug mode (if it relies on throwing an exception).

**Review by s-barannikov — APPROVED** [review]
LGTM

File: compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp
Comment (seanshpark) [reply compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp:609]: I'll wait for this before merge...

Comment (jinevening) [reply compiler/luci/pass/src/QuantizeWithMinMaxPass.cpp]: > It could also break a negative test in debug mode

@s-barannikov  Agreed. I've removed the check. PTAL

**Review by seanshpark — APPROVED** [review]
LGTM

**Review by s-barannikov — APPROVED** [review]

**Pull request merged**
